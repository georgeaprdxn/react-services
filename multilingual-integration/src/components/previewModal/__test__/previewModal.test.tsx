import React from 'react'
import { shallow } from 'enzyme'

import mockPreviewModalData from './previewModal.mock.json'
import PreviewOverlay from '../example/previewModal.example'
import PreviewModal from '../index'

describe('PreviewModal component tests', () => {
	let PreviewModalComp: any = null

	beforeEach(() => {
		PreviewModalComp = shallow(<PreviewOverlay />)
	})

	it('should render a button', () => {
		expect(PreviewModalComp.find('.custom-button').length).toEqual(1)
	})

	it('calls open modal', () => {
		PreviewModalComp.find('.custom-button').simulate('click')
		expect(PreviewModalComp.state('showModal')).toBe(true)
	})

	it('validate child component props', () => {
		PreviewModalComp.find('.custom-button').simulate('click')

		expect(PreviewModalComp.find(PreviewModal).prop('width')).toEqual(
			mockPreviewModalData.width
		)
		expect(PreviewModalComp.find(PreviewModal).prop('maxWidth')).toEqual(
			mockPreviewModalData.maxWidth
		)
		expect(
			PreviewModalComp.find(PreviewModal).prop('disableCloseAction')
		).toEqual(mockPreviewModalData.disableCloseAction)
	})

	it('calls close modal', () => {
		PreviewModalComp.instance().closeModal()
		expect(PreviewModalComp.state('showModal')).toBe(false)
	})

	it('matches the snaphot', () => {
		expect(PreviewModalComp).toMatchSnapshot()
	})
})
