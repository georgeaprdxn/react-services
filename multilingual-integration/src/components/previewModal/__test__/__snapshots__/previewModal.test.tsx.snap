// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewModal component tests matches the snaphot 1`] = `
<Fragment>
  <button
    className="custom-button"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    Open Modal
  </button>
</Fragment>
`;
